src/utils/codec.ts

```
@@ -397,6 +428,138 @@ function reshapeArray(flat: unknown[], shape: readonly number[]): unknown {
+    return 1;
+  }
+  return value.reduce((count, item) => count + countJsonLeaves(item), 0);
+}
+
+function assertJsonMatchesShape(
+  data: unknown,
+  shape: readonly number[],
+  dtype: string | undefined
+): void {
+  const declaredCount = shapeProduct(shape);
+  const strictLeaves = /^(?:bool|u?int\d*|float\d*)$/.test(dtype ?? '');
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)